tests/pos/i1747.scala

```diff
@@ -0,0 +1,3 @@
+abstract class Coll[E] extends java.util.Collection[E] {
+  def toArray[T](a: Array[T]): Array[T] = ???
```

Comment:
Additionally, the `pos` test is not testing anything as it is an abstract class and it could be defining new abstract members.
```scala
abstract class Coll[E] extends java.util.Collection[E] {
  def toArray[T](a: Array[T]): Array[T] = ???
}
```

This is in fact generating the following method in the bytecode.

```scala
public Object toArray(Object a) { ... }
```
I added run tests that reproduce the issue (the CI will fail).